### ISSUE TITLE
buyProduct not returning E_USER_CANCELLED when cancelling a transaction

### ISSUE THREAD

**Issue by kennym** [dooboolab/react-native-iap#583]
### Version of react-native-iap
3.2.3
### Version of react-native
0.58.6
### Platforms you faced the error (IOS or Android or both?)
iOS
### Expected behavior
When the user cancels a transaction, `RNIap.buyProduct` returns `null` instead of raising an exception with `error.code` `E_USER_CANCELLED`.
### Actual behavior
When the user cancels a transaction, `RNIap.buyProduct` returns `null`
### Tested environment (Emulator? Real Device?)
Real device
### Steps to reproduce the behavior

This is the code:
```
 async buyItem(sku) {
    try {
      const purchase = await RNIap.buyProduct(sku)

      if (!purchase) {
        // This should not happen
        return
      }

      // Omitted code
    } catch (err) {
      if (err.code && err.code === 'E_USER_CANCELLED') {
        // Not being called
      }
    }
  }
```

1. Tap on a product on device
2. Wait for subscription confirmation dialog to appear.
3. Click Cancel

**Comment by hyochan** [comment]
@kennym I've seen related issue earlier and we need to fix this. Currently, cancelling send @null array in `ios`. I hope you can give us a `PR`.

**Comment by solidfox** [comment]
You can (and probably should) use the new listener pattern to listen to errors. An error with `responseCode` `1` on android and `"2"` on iOS means the user cancelled. Note that iOS will give you the code as a string while Android enters it as an integer.

**Comment by Rinkuk1993** [comment]
@solidfox I think it should be handled in plugin as @hyochan suggested. The plugin shall send the same error code for both Android and iOS in the updated listener pattern.
I will see if I get time to add PR to this. But before this let me know if it is ok to override code and message to what ```rejectPromiseWithBillingError()``` is generating.

**Comment by leethree** [comment]
The flow and TypeScript type of `responseCode` is `number`, but it's string on iOS 🤷‍♂

**Comment by hyochan** [comment]
Closed via #616
